src/Microsoft.Windows.CsWin32/templates/SYSTEMTIME.cs

```diff
@@ -0,0 +1,75 @@
+﻿partial struct SYSTEMTIME
+	: IEquatable<SYSTEMTIME>
+{
+	public bool Equals(SYSTEMTIME other)
+	{
+		return this.wYear == other.wYear &&
+			   this.wMonth == other.wMonth &&
+			   this.wDayOfWeek == other.wDayOfWeek &&
+			   this.wDay == other.wDay &&
+			   this.wHour == other.wHour &&
+			   this.wMinute == other.wMinute &&
+			   this.wSecond == other.wSecond &&
+			   this.wMilliseconds == other.wMilliseconds;
+	}
+
+	public override bool Equals(object obj) => obj is SYSTEMTIME other && this.Equals(other);
+	public override int GetHashCode() => (this.wYear, this.wMonth, this.wDayOfWeek, this.wDay, this.wHour, this.wMinute, this.wSecond, this.wMilliseconds).GetHashCode();
+
+	public static bool operator ==(SYSTEMTIME d1, SYSTEMTIME d2)
+	{
+		return d1.wYear == d2.wYear &&
+			   d1.wMonth == d2.wMonth &&
+			   d1.wDayOfWeek == d2.wDayOfWeek &&
+			   d1.wDay == d2.wDay &&
+			   d1.wHour == d2.wHour &&
+			   d1.wMinute == d2.wMinute &&
+			   d1.wSecond == d2.wSecond &&
+			   d1.wMilliseconds == d2.wMilliseconds;
+	}
+
+	public static bool operator !=(SYSTEMTIME d1, SYSTEMTIME d2)
+	{
+		return d1.wYear != d2.wYear &&
+			   d1.wMonth != d2.wMonth &&
+			   d1.wDayOfWeek != d2.wDayOfWeek &&
+			   d1.wDay != d2.wDay &&
+			   d1.wHour != d2.wHour &&
+			   d1.wMinute != d2.wMinute &&
+			   d1.wSecond != d2.wSecond &&
+			   d1.wMilliseconds != d2.wMilliseconds;
+	}
+
+	public static explicit operator global::System.DateTime(SYSTEMTIME sysTime)
+	{
+		if (sysTime == default(SYSTEMTIME) || sysTime.wYear <= 0 || sysTime.wMonth <= 0 || sysTime.wDay <= 0)
+		{
+			return default;
+		}
+
+		// DateTime gets DayOfWeek automatically
+		return new global::System.DateTime(sysTime.wYear,
+			sysTime.wMonth, sysTime.wDay, sysTime.wHour,
+			sysTime.wMinute, sysTime.wSecond, sysTime.wMilliseconds);
+	}
+
+	public static explicit operator SYSTEMTIME(global::System.DateTime time)
```

Comment:
This could maybe be implicit. But if it is ever used in interop it might cause issues because the minimum value for year is `1601`.
https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-systemtime

Reply:
Let's leave it as explicit for now. It's safer to move from explicit to implicit later than the other way around.